kernel/names.mli

```diff
@@ -643,8 +643,17 @@ module Projection : sig
   val print : t -> Pp.t
   (** Print internal representation (not to be used for user-facing messages). *)
 
+  val debug_to_string : t -> string
+  (** Same as [to_string], but outputs extra information related to debug. *)
+
+  val debug_print : t -> Pp.t
+  (** Same as [print], but outputs extra information related to debug. *)
+
 end
 
+(** Predicate on projection representation (ignoring unfolding state) *)
+module PRpred : Predicate.S with type elt = Projection.t
```

Comment:
I would rather have
```suggestion
module PRpred : Predicate.S with type elt = Projection.Repr.t
```
Repr.t is the projection without the unfolding state

Reply:
Then do I switch to `is_transparent_projection : t -> Projection.Repr.t -> bool` in the `TransparentState` module?
Or do you prefer that the function internally calls `Projection.repr`?

Reply:
>Then do I switch to

I think so

Then RedFlags.red_projection still takes Projection.t and returns unfolded || in the transparent state

Reply:
I did that, I think it makes more sense with what you suggested.